src/EnergyPlus/PondGroundHeatExchanger.cc

```diff
@@ -443,16 +441,17 @@ namespace PondGroundHeatExchanger {
             this->MyFlag = false;
         }
 
-        // check if we are in very first call for this zone time step
-        if (DataGlobals::BeginTimeStepFlag && FirstHVACIteration &&
-            DataPlant::PlantLoop(this->LoopNum).LoopSide(this->LoopSideNum).FlowLock == 1) { // DSU
+        if (FirstHVACIteration && !DataHVACGlobals::ShortenTimeStepSys && this->firstTimeThrough) {
```

Comment:
So this only saves the PastBulkTemperature at the zone time step yet the pond HX is simulated at the system time step. Makes me think of the zone heat balance history terms that are adjusted when the system time step is reduced.

Reply:
@rraustad the idea is to advance time once per system time step.  That is confirmed with the time step used in the update equation [here](https://github.com/NREL/EnergyPlus/blob/c6c1fcb9418c83bd273d0adf567a1f524fee3db0/src/EnergyPlus/PondGroundHeatExchanger.cc#L520).  The IF side of this should only be hit one time per system time step, right at the beginning.  This will ensure that the temperatures are shifted from the last time step once, right then.  It was wrong before, and I think this new way is closer, but I'm having trouble keeping all the flags straight, so I may need to run an experiment to convince myself.  

@mitchute we may need to run an experiment where we add a report variable to the pond heat exchanger that reports how many times it thinks it updated the temperatures per system time step.  The value can be updated in the context of this logic and then if we report it out at the detailed interval, it *should* be exactly 1 every system time step.

Reply:
@myoldmopar I added your suggested logic on the ```ELSE``` statement and things went all sorts of haywire. I'll add this report variable with the current logic and report back.

Reply:
A perfect test to see if this is working correctly.

Reply:
@rraustad @Myoldmopar I added a report variable as you suggested which incremented by 1 upon entering the ```IF``` section noted here. As is, the ```IF``` block is being hit multiple times per iteration, as you expected. Changing ```ELSE``` to ``` ELSE IF NOT FIRSTHVAC``` causes the temperature to be updated exactly 1 time per system timestep. The results looked bad when compared to the develop, but I think this gives us some confidence that at least this piece is working as expected.

![Var_42](https://user-images.githubusercontent.com/2985979/81338370-c20fc280-9069-11ea-9759-fccf30a654cf.png)
![Var_0](https://user-images.githubusercontent.com/2985979/81338634-3185b200-906a-11ea-91c2-24ae341aa061.png)
![Var_0](https://user-images.githubusercontent.com/2985979/81338701-4bbf9000-906a-11ea-91ad-f065c718fa50.png)

Reply:
Thanks for the update on the report variable, I was glad to see it going up by exactly 1 every single system timestep in detailed output.  That was very reassuring.  I think this is an exceedingly important change.  The pond was artificially advancing the temperatures through time way, way, more than it should've been.  This new pattern of checking for a new time step can be applied throughout the hvac and plant world as a reliable way to move through time.  I don't like these diffs, but as far as I can tell, this is much, much more correct than before.